src/open_inwoner/openklant/wrap.py

```diff
@@ -88,211 +82,42 @@ def fetch_klantcontactmoment(
-        return []
-
-    object_contact_momenten = factory(ObjectContactMoment, response)
-
-    # resolve linked resources
-    object_mapping = {}
-    for ocm in object_contact_momenten:
-        assert ocm.contactmoment == contactmoment.url
-        ocm.contactmoment = contactmoment
-        if ocm.object_type == "zaak":
-            object_url = ocm.object
-            # Avoid fetching the same object, if multiple relations wit the same object exist
-            if ocm.object in object_mapping:
-                ocm.object = object_mapping[object_url]
-            else:
-                ocm.object = fetch_case_by_url_no_cache(ocm.object)
-                object_mapping[object_url] = ocm.object
-
-    return object_contact_momenten
-
-
-def fetch_objectcontactmomenten_for_object_type(
-    contactmoment: ContactMoment, object_type: str
-) -> List[ObjectContactMoment]:
-    client = build_client("contactmomenten")
-
-    moments = _fetch_objectcontactmomenten_for_contactmoment(
-        contactmoment, client=client
-    )
-
-    # eSuite doesn't implement a `object_type` query parameter
-    ret = [moment for moment in moments if moment.object_type == object_type]
-
-    return ret
-
-
 def fetch_objectcontactmoment(
     contactmoment: ContactMoment, object_type: str
 ) -> Optional[ObjectContactMoment]:
-    ocms = fetch_objectcontactmomenten_for_object_type(contactmoment, object_type)
-    if ocms:
-        return ocms[0]
-
-
-def _fetch_klanten_for_bsn(user_bsn: str, *, client=None) -> List[Klant]:
-    client = client or build_client("klanten")
-    if client is None:
-        return []
-
-    try:
-        response = get_paginated_results(
-            client,
-            "klanten",
-            params={"subjectNatuurlijkPersoon__inpBsn": user_bsn},
-        )
-    except (RequestException, ClientError) as e:
-        logger.exception("exception while making request", exc_info=e)
-        return []
-
-    klanten = factory(Klant, response)
-
-    return klanten
-
-
-def _fetch_klanten_for_kvk_or_rsin(
-    user_kvk_or_rsin: str, *, vestigingsnummer=None, client=None
-) -> List[Klant]:
-    client = client or build_client("klanten")
+    client = build_client("contactmomenten")
     if client is None:
-        return []
-
-    params = {"subjectNietNatuurlijkPersoon__innNnpId": user_kvk_or_rsin}
-
-    if vestigingsnummer:
-        params = {
-            "subjectVestiging__vestigingsNummer": vestigingsnummer,
-        }
-
-    try:
-        response = get_paginated_results(
-            client,
-            "klanten",
-            params=params,
-        )
-    except (RequestException, ClientError) as e:
-        logger.exception("exception while making request", exc_info=e)
-        return []
-
-    klanten = factory(Klant, response)
+        return
 
-    return klanten
+    return client.retrieve_objectcontactmoment(contactmoment, object_type)
 
 
 def fetch_klant(
```

Comment:
I'm not entirely sure if we still need these functions after the changes, I suppose they do save us the hassle of having to build a client when we use it in views

Reply:
I think we could remove them and just use the client instances in the views. 

These are a level of indirection and we should really be re-using clients (for connection and expensive cert/schema setups etc).

Reply:
I'll remove them :+1:

Reply:
I removed these functions and now use the clients directly. I wanted to switch to using context managers (`with build_client("zaak")`, but in some cases we can't do this currently, because we do not raise errors if we can't build a client but return `None` instead. This is related to the issue that we currently simply show no results if we can't build a client